tests/runtime/kubernetes/test_kubernetes_executor.py

```diff
@@ -88,6 +88,9 @@ def monitor_task(self, host, execution_date, dag_id, task_id, expected_final_sta
                             execution_date=execution_date,
                             task_id=task_id)
                 )
+                if result.status_code == 404:
+                    check_call(["echo", "api returned 404."])
```

Comment:
What is this for?

Quoting from slack too from Matt: 
>One thing I thought of, correct me if I’m wrong, is there a possibility for an infinite loop? Say the status always returns 404, tries also would not increment. Maybe put the tries increment at the beginning of the loop and increase the max_tries?

Reply:
https://apache-airflow.slack.com/archives/CCPRP7943/p1582572265058300?thread_ts=1582567409.032300&cid=CCPRP7943

Reply:
Ah yeah! tries += 1 is missing. Great catch